### PR TITLE
Handle 'none' IP Address on nxos_ssh for get_interfaces_ip

### LINKED ISSUE #1020: NXOS_SSH: get_interfaces_ip fails with KeyError when interface have no IP addr assigned to it.
### Description of Issue/Question
Calling get_interfaces_ip with nxos_ssh on a devices containing Interfaces without IP assigned to it, it fails with a key error.

*Note*: Please check https://guides.github.com/features/mastering-markdown/
to see how to properly format your request.

### Did you follow the steps from https://github.com/napalm-automation/napalm#faq
(Place an ``x`` between the square brackets where applicable)

- [ x] Yes
- [ ] No


### Setup

### napalm version
(Paste verbatim output from `pip freeze | grep napalm` between quotes below)

```
napalm==2.4.0
```

### Network operating system version
(Paste verbatim output from `show version` - or equivalent - between quotes below)

```
Cisco Nexus Operating System (NX-OS) Software
TAC support: http://www.cisco.com/tac
Copyright (C) 2002-2019, Cisco and/or its affiliates.
All rights reserved.
The copyrights to certain works contained in this software are
owned by other third parties and used and distributed under their own
licenses, such as open source.  This software is provided "as is," and unless
otherwise stated, there is no warranty, express or implied, including but not
limited to warranties of merchantability and fitness for a particular purpose.
Certain components of this software are licensed under
the GNU General Public License (GPL) version 2.0 or 
GNU General Public License (GPL) version 3.0  or the GNU
Lesser General Public License (LGPL) Version 2.1 or 
Lesser General Public License (LGPL) Version 2.0. 
A copy of each such license is available at
http://www.opensource.org/licenses/gpl-2.0.php and
http://opensource.org/licenses/gpl-3.0.html and
http://www.opensource.org/licenses/lgpl-2.1.php and
http://www.gnu.org/licenses/old-licenses/library.txt.

Software
  BIOS: version 07.65
  NXOS: version 7.0(3)I7(6)
  BIOS compile time:  09/04/2018
  NXOS image file is: bootflash:///nxos.7.0.3.I7.6.bin
  NXOS compile time:  3/5/2019 13:00:00 [03/05/2019 22:04:55]


Hardware
  cisco Nexus9000 C93108TC-EX chassis 
  Intel(R) Xeon(R) CPU  @ 1.80GHz with 24633600 kB of memory.
  Processor Board ID FDO21332HC8

  Device name: ams1dc1-net-f02lf17.ams1
  bootflash:   53298520 kB
Kernel uptime is 53 day(s), 21 hour(s), 14 minute(s), 55 second(s)

Last reset at 770279 usecs after Thu May 23 15:07:16 2019
  Reason: Power Down/UP epld upgrade process
  System version: 7.0(3)I7(6)
  Service: Power Down/UP epld upgrade process

plugin
  Core Plugin, Ethernet Plugin

Active Package(s):

```

### Steps to Reproduce the Issue
Call get_interfaces_ip with nxos_ssh driver on a device containing at least 1 interface without an IP address assigned to it.

### Error Traceback

```
$ napalm -v nxos_ssh --user itifous --password ***** n9k-hostname call get_interfaces_ip
2019-07-16 13:51:56,644 - napalm - ERROR - method - Failed: list index out of range

================= Traceback =================

Traceback (most recent call last):
  File "/home/itifous/.local/share/virtualenvs/netbox-netprod-importer-1U3vzfIV/bin/napalm", line 11, in <module>
    load_entry_point('napalm', 'console_scripts', 'napalm')()
  File "/home/itifous/.local/share/virtualenvs/netbox-netprod-importer-1U3vzfIV/lib/python3.7/site-packages/napalm-2.4.0-py3.7.egg/napalm/base/clitools/cl_napalm.py", line 309, in main
    run_tests(args)
  File "/home/itifous/.local/share/virtualenvs/netbox-netprod-importer-1U3vzfIV/lib/python3.7/site-packages/napalm-2.4.0-py3.7.egg/napalm/base/clitools/cl_napalm.py", line 292, in run_tests
    call_getter(device, args.method, **method_kwargs)
  File "/home/itifous/.local/share/virtualenvs/netbox-netprod-importer-1U3vzfIV/lib/python3.7/site-packages/napalm-2.4.0-py3.7.egg/napalm/base/clitools/cl_napalm.py", line 31, in wrapper
    r = func(*args, **kwargs)
  File "/home/itifous/.local/share/virtualenvs/netbox-netprod-importer-1U3vzfIV/lib/python3.7/site-packages/napalm-2.4.0-py3.7.egg/napalm/base/clitools/cl_napalm.py", line 256, in call_getter
    r = func(**kwargs)
  File "/home/itifous/.local/share/virtualenvs/netbox-netprod-importer-1U3vzfIV/lib/python3.7/site-packages/napalm-2.4.0-py3.7.egg/napalm/nxos_ssh/nxos_ssh.py", line 892, in get_interfaces_ip
    prefix_len = int(line.split()[5].split("/")[1])
IndexError: list index out of range

```
Cause by : 
``` 
Vlan1003, Interface status: protocol-up/link-up/admin-up, iod: 3,
  IP address: none
  IP broadcast address: 255.255.255.255
  IP multicast groups locally joined: none
  IP MTU: 1500 bytes (using link MTU)
  IP proxy ARP : disabled
  IP Local Proxy ARP : disabled
  IP multicast routing: disabled
  IP icmp redirects: disabled
  IP directed-broadcast: disabled 
  IP Forwarding: enabled 
  IP icmp unreachables (except port): disabled
  IP icmp port-unreachable: enabled
  IP unicast reverse path forwarding: none
  IP load sharing: none 
  IP interface statistics last reset: never
  IP interface software stats: (sent/received/forwarded/originated/consumed)
......
.....
```

The method get_interfaces_ip does not check against a 'none' string before processing the IP address : 
```ipdb> list                                                                                                                                                                                                                                                                                             
    889             if "IP address" in line:
    890                 ip_address = line.split(",")[0].split()[2]
    891                 try:
 -->   892                     prefix_len = int(line.split()[5].split("/")[1])
    893                 except ValueError:
    894                     prefix_len = "N/A"
    895                 val = {"prefix_length": prefix_len}
    896                 v4_interfaces.setdefault(interface, {})[ip_address] = val
    897 
    898         v6_interfaces = {}
    899         for line in output_v6.splitlines():
```

```> /home/itifous/.local/share/virtualenvs/netbox-netprod-importer-1U3vzfIV/lib/python3.7/site-packages/napalm-2.4.0-py3.7.egg/napalm/nxos_ssh/nxos_ssh.py(892)get_interfaces_ip()
    891                 try:
--> 892                     prefix_len = int(line.split()[5].split("/")[1])
    893                 except ValueError:

ipdb> line                                                                                                                                                                                                                                                                                             
'  IP address: none'
   ```

I think simply check against the 'none' str could fix the issue, I will file a PR.

### PR DESCRIPTION
**Pull request by fousti**
Fixes #1020

### CODE REVIEW

**Comment by bewing** [comment]
I spot-checked some other drivers for behavior.

EOSDriver:  Lists routed interfaces with no ipv4/ipv6 with empty dictionary:
```
{'Ethernet1': {'ipv4': {}, 'ipv6': {}}}
```
IOSDriver:  Completely omits  routed interfaces with no IPs from output
JunOSDriver:  Omits routed interfaces with no IPs

Would probably want to use one of the above two methods of indicating lack of IP, rather than returning textual data.

**Comment by fousti** [comment]
Sounds better indeed, i'll get this done ASAP.

**Comment by fousti** [comment]
Done :)

**Comment by fousti** [comment]
@bewing Do you think this could be review again please ?

File: napalm/base/test/models.py [hunk 1]
```diff
@@ -174,7 +174,7 @@
     "jitter": float,
 }
 
-interfaces_ip = {"prefix_length": int}
+interfaces_ip = {"prefix_length": (int, text_type)}
```
Comment (bewing) [inline napalm/base/test/models.py]: Don't think this line change is needed anymore

**Comment by mirceaulinic** [comment]
@fousti are you still working on this?

**Comment by fousti** [comment]
@mirceaulinic Yes, I have rebased & change the test model according to @bewing suggestions.

**Comment by fousti** [comment]
Hi @mirceaulinic , when do you think this will be merged ?
Thanks

**Pull request merged**